checks/__init__.py

```diff
@@ -560,6 +607,14 @@ def run(self):
                     tb=traceback.format_exc()
                 )
             instance_statuses.append(instance_status)
+
+        if self.in_developer_mode:
+            try:
+                after = AgentCheck._collect_stats()
+                self.set_stats(before, after)
+            except Exception: # It's fine if we can't collect stats for the run
+                pass
+
```

Comment:
Here I am not sure if for instance the CPU percentage sampling before and after would be useful (same for IO) because we would need to see it _during_ the call of `.check()`
For memory, we could see the leaks I guess.